test/integration/compile-swagger-test.coffee

```diff
@@ -97,25 +97,44 @@ describe('compile() · Swagger', ->
   )
 
   describe('with multiple responses', ->
-    # Multiple responses is the closest we can get with Swagger to something
-    # like transaction examples ¯\_(ツ)_/¯
-
+    transactions = undefined
+    filename = 'apiDescription.json'
     detectTransactionExamples = sinon.spy(require('../../src/detect-transaction-examples'))
-    compilationResult = undefined
-    transaction = undefined
+
+    expectedStatusCodes = [200, 400, 500]
 
     beforeEach((done) ->
       stubs = {'./detect-transaction-examples': detectTransactionExamples}
 
-      compileFixture(fixtures.multipleResponses.swagger, {stubs}, (args...) ->
+      compileFixture(fixtures.multipleResponses.swagger, {filename, stubs}, (args...) ->
         [err, compilationResult] = args
-        transaction = compilationResult.transactions[0]
+        transactions = compilationResult?.transactions
         done(err)
       )
     )
 
     it('detection of transaction examples was not called', ->
       assert.isFalse(detectTransactionExamples.called)
     )
+    it('expected number of transactions was returned', ->
+      assert.equal(transactions.length, expectedStatusCodes.length)
+    )
+
+    for statusCode, i in expectedStatusCodes
+      do (statusCode, i) ->
+        context("origin of transaction ##{i + 1}", ->
+          it('uses URI as resource name', ->
+            assert.equal(transactions[i].origin.resourceName, '/honey')
```

Comment:
`transactions` should™ be `undefined` only in case `compileFixture` passes an error to the callback. In that case, the `transactions = compilationResult?.transactions` line will pass, but subsequent `done(err)` line will propagate the error and stop the test. It won't ever get to line 127 in that case.